### ISSUE TITLE
How to run PyTorch BERT pre-training on a single GPU?

### ISSUE THREAD

**Issue by nehcixsjc** [NVIDIA/DeepLearningExamples#260]
Is there a simple recipe to run PyTorch BERT pre-training on a single V100? I tried to modify the `num_gpus` to 1 in `run_pretraining.sh` and then ran into the following error

```
10/17/2019 17:16:01 - INFO - __main__ -   device cuda n_gpu 1 distributed training False
10/17/2019 17:16:12 - INFO - __main__ -   ***** Running training *****
10/17/2019 17:16:12 - INFO - __main__ -     Batch size = 64
Selected optimization level O2:  FP16 training with FP32 batchnorm and FP32 master weights.

Defaults for this optimization level are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : True
master_weights         : True
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : True
master_weights         : True
loss_scale             : dynamic
SEED 28803
  LR =  0.006
Training. . .
Traceback (most recent call last):
  File "/workspace/bert/run_pretraining.py", line 578, in <module>
    args = main()
  File "/workspace/bert/run_pretraining.py", line 464, in main
    data_file = files[(f_start_id*torch.distributed.get_world_size()+torch.distributed.get_rank())%num_files]
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 584, in get_world_size
    return _get_group_size(group)
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 200, in _get_group_size
    _check_default_pg()
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 191, in _check_default_pg
    "Default process group is not initialized"
AssertionError: Default process group is not initialized
```

**Comment by sharathts** [comment]
Thanks for bringing it up. It will be fixed in https://github.com/NVIDIA/DeepLearningExamples/pull/261

Note: When training with different number of gpus, make sure num_gpus * gradient_accumulation_steps * batch_size = 65536 for phase 1 and 32768 for phase 2 respectively.

**Comment by nehcixsjc** [comment]
Thanks for fixing it, @sharathts !

**Comment by MaveriQ** [comment]
> Thanks for bringing it up. It will be fixed in #261
> 
> Note: When training with different number of gpus, make sure num_gpus * gradient_accumulation_steps * batch_size = 65536 for phase 1 and 32768 for phase 2 respectively.

In the default script, we have 
n_gpu=8
batch_size=8192
gradient_accumulation_steps=128

That doesn't add up to 65536 for phase 1

**Comment by sharathts** [comment]
Script internally divides train_batch_size specified on the command line by gradient_accumulation steps.
